File: tests/testers/test-html/cgi-bin/method-not-allowed/stdin.py
#!/usr/bin/python3
import sys
import select

timeout = 3  # Timeout en secondes
ready, _, _ = select.select([sys.stdin], [], [], timeout)
print("Content-type: text/html\r\n")
print("<html>")
print("<head>")
print("<title>CGI Test Python - stdin.py</title>")
print("</head>")
print("<body>")
print("<h1>CGI Test Python - stdin.py</h1><p>")
if ready:
    input_data = sys.stdin.read()
    if input_data == "":
        print("Server has no data to send")
    else:
        print("Server sent data: ", input_data)
else:
    print("Server has sent no data received within", timeout, "seconds.")
print("<p></body>")
print("</html>")

File: tests/testers/test-html/cgi-bin/pl/stdin.pl
#!/usr/bin/perl

use strict;
use warnings;
use IO::Select;

my $timeout = 3;
my $selector = IO::Select->new();
$selector->add(\*STDIN);

print "Content-type: text/html\n\n";
print "<html>\n";
print "<head>\n";
print "<title>CGI Test Perl - stdin.pl</title>\n";
print "</head>\n";
print "<body>\n";
print "<h1>CGI Test Perl - stdin.pl</h1><p>\n";

if (my @ready = $selector->can_read($timeout)) {
    my $input_data = <STDIN>;
    if (!defined $input_data || $input_data eq "") {
        print "Server has no data to send\n";
    } else {
        print "Server sent data: $input_data\n";
    }
} else {
    print "Server has sent no data received within $timeout seconds.\n";
}

print "<p></body>\n";
print "</html>\n";
